data_store/cli.py

```diff
@@ -18,148 +23,276 @@
+
+    Example usage:
+        flask admin add-roles-to-users --filepath <path> --roles <role1>,<role2>
+    """
+
+    result = {
+        "created": 0,
+        "updated": 0,
+        "errors": {},
+    }
+
+    with current_app.app_context():
+        with click.open_file(filepath) as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                email = row["email"]
+                full_name = row["full_name"]
+
+                try:
+                    emailinfo = validate_email(email, check_deliverability=False)
+                    email = emailinfo.normalized
+
+                except EmailNotValidError as e:
+                    result["errors"][email] = e.response.text
+                    click.echo(f"{email} error: {str(e)}")
+
+                    # Go to next email
+                    continue
+
+                # Try to get account by email
+                try:
+                    response = requests.get(
+                        f"{Config.ACCOUNT_STORE_API_HOST}{Config.ACCOUNTS_ENDPOINT}",
+                        headers={"Content-Type": "application/json"},
+                        params={"email_address": email},
+                    )
+
+                    response.raise_for_status()
+
+                    account = response.json()
+                except RequestException as e:
+                    if e.response.status_code != 404:
+                        result["errors"][email] = e.response.text
+                        click.echo(f"Get account with {email} error: {e.response.text}")
+
+                        # Go to next email
+                        continue
+
+                    # If account not found, set account to None and we create it
+                    account = None
+
+                if account is None:
+                    click.echo(f"Creating account for {email}")
+
+                    # Create account
+                    try:
+                        response = requests.post(
+                            f"{Config.ACCOUNT_STORE_API_HOST}{Config.ACCOUNTS_ENDPOINT}",
+                            headers={"Content-Type": "application/json"},
+                            json={"email_address": email},
+                        )
+
+                        response.raise_for_status()
+
+                        result["created"] += 1
+                        account = response.json()
+                    except requests.exceptions.RequestException as e:
+                        result["errors"][email] = e.response.text
+                        click.echo(f"Create account for {email} error: {e.response.text}")
+
+                        # Go to next email
+                        continue
+
+                # Update account with roles
+                data = {
+                    "email_address": email,
+                    "azure_ad_subject_id": account.get("azure_ad_subject_id"),
+                    "full_name": full_name,
+                    "roles": roles,
```

Comment:
This doesn't add to the roles - it overwrites them. Is that what you're expecting to do? I think we discussed making it the combination of roles rather than overwriting.

If it removes then ... maybe `set-roles-for-users` **is** better than `add-roles-to-users`?

Reply:
So.. I think we can do it either way, especially now that this script will be used mainly for testing purposes.

I think setting the roles would be useful in more circumstances, as it deletes and creates the roles. Adding does not enable us to remove existing roles.

What do you think?

Reply:
I think setting is fine - but maybe it should echo the roles that are removed just so there's some chance of catching a mistake?

Reply:
Changed name back to `set-roles-for-users`. Roles are overwritten. Now it echoes roles that will be removed.